### PR TITLE
Added xlabel= kwarg to mpf.plot() 

### PR DESCRIPTION
**Pull request by vedant-gawande**
Please merge this repo as you can see i have done the work as per requirement

### CODE REVIEW

**Comment by DanielGoldfarb** [comment]
@vedant-gawande 
Ved,
Thanks so much for contributing.  At a glance it looks good.  I'm short on time for this project today, but will look into merging this early next week.  All the best.  --Daniel

**Comment by DanielGoldfarb** [comment]
issue #427

**Comment by vedant-gawande** [comment]
> @vedant-gawande Ved, Thanks so much for contributing. At a glance it looks good. I'm short on time for this project today, but will look into merging this early next week. All the best. --Daniel

Yeah sure sir,
No problem

File: src/mplfinance/plotting.py [hunk 1]
```diff
@@ -184,6 +184,10 @@ def _valid_plot_kwargs():
         'axtitle'                   : { 'Default'     : None, # Axes Title (subplot title)
                                         'Description' : 'Axes Title (subplot title)',
                                         'Validator'   : lambda value: isinstance(value,(str,dict)) },
+                                        
+        'xlabel'                    : { 'Default'     : 'Date', # x-axis label
+                                        'Description' : 'label for x-axis of main plot',
+                                        'Validator'   : lambda value: isinstance(value,str) },
```
Comment (DanielGoldfarb) [inline src/mplfinance/plotting.py:190]: A block of code, similar to this, needs to be also added to `_valid_addplot_kwargs()` ... approximately [**here**](https://github.com/matplotlib/mplfinance/blob/master/src/mplfinance/plotting.py#L1300-L1301).

File: src/mplfinance/plotting.py [hunk 2]
```diff
@@ -1070,6 +1075,8 @@ def _addplot_columns(panid,panels,ydata,apdict,xdates,config):
 def _addplot_apply_supplements(ax,apdict,xdates):
     if (apdict['ylabel'] is not None):
         ax.set_ylabel(apdict['ylabel'])
+    if (apdict['xlabel'] is not None):
```
Comment (DanielGoldfarb) [inline src/mplfinance/plotting.py:1078]: This will not work until the new kwarg is ***also*** added to `_valid_addplot_kwargs()` (as noted above).
Thanks.

**Review by DanielGoldfarb — CHANGES_REQUESTED** [review]
see comments.  please make minor change requested and push to your fork (will automatically propagate to the PR).

**Comment by DanielGoldfarb** [comment]
It turns out that apparently this was not tested at all.   A little bit of testing showed that it was not working for all cases where the main plot is *not* the bottom-most panel.   
1. In panels mode, the `xlabel` should always be on the bottom panel, and it makes no sense for addplot to support xlabel (since it will always be on the bottom-most panel).  Therefore moved xlabel implementation into `_set_ticks_on_bottom_panel_only()` when in panels mode.  
2. In external axes mode, the user can always call `Axes.set_xlabel()` directly on whichever axes they choose, so again no need to support xlabel in addplot.

This changes were made and merged into the PR.

@vedant-gawande   Thank you for contributing to mplfinance!   Wishing you all the best.

**Pull request merged**